app/models/season/phase_switcher.rb

```diff
@@ -1,8 +1,11 @@
-module SeasonPhaseSwitcher
-  extend ActiveSupport::Concern
+class Season::PhaseSwitcher
+
+  def initialize(season)
```

Comment:
Oh, and when the `@season` ivar is gone, it's questionable if we still need an _instance_ of `Season::PhaseSwitcher` at all; there is no state left and all the methods are self-contained. We can make all the methods class methods.